encode.go

```diff
@@ -53,6 +53,8 @@ type Encoder struct {
 	w   writer
 	buf []byte
 
+	tbuf []byte
```

Comment:
Please group this with other `buf` var and rename to `timeBuf`. Also having a comment why we need another buffer ("We can't use use e.buf, as it's used to encode extLen") will help future readers a lot.

Reply:
thanks, updated